survey_logic/report.py

```diff
@@ -0,0 +1,124 @@
+from fastapi import Depends
+from dependency_injector.wiring import Provide, inject
+from typing import Optional
+import plotly.express as px
+import pandas as pd
+
+from utils.html_report import HTMLReport
+from repository.sqlite_repository import SQLiteRepository
+from repository.yaml_rule_repository import YamlRulesRepository
+from utils.container import Container
+
+@inject
+async def generate_project_report(
+    sqlite_repo: SQLiteRepository = Depends(Provide[Container.sqlite_repo]),
+    rulesYamlConfig: YamlRulesRepository = Depends(Provide[Container.rules_config]),
+) -> str:
+    """
+    Generates a report for all projects with different statistics
+
+    Returns:
+        str: The report in HTML format
+    """
+    html_repository = HTMLReport(reportFile="surveyReport.html")
+
+    projects = []
+
+    for project_name in rulesYamlConfig.getProjectNames():
+        project = await sqlite_repo.get_project_by_name(project_name)
+        average_rating = sqlite_repo.get_project_avg_rating(project.id)
+        comments_number = sqlite_repo.get_number_of_comment(project.id)
+        display_modal_number = sqlite_repo.get_number_of_display(project.id)
+        feature_urls = await sqlite_repo.get_features_urls_by_project_name(project_name)
+        active_rules = [
+            rule
+            for rule in rulesYamlConfig.getRulesFromProjectName(project_name)
+            if rule.is_active == True
```

Comment:
The condition `if rule.is_active == True` can be simplified to `if rule.is_active`.

Reply:
It wasn't my code, I just moved it, but changed :)